### ISSUE TITLE
Product tours / guides

### ISSUE THREAD

**Issue by annikaschmid** [PostHog/posthog#25726]
## Is your feature request related to a problem?

We have a in-app surveys product, that allows our customers to show a survey to their users based on specific triggers, such as properties, events, or page views.

A product tours product would work in a similar way, and enable our customers to create no-code or low code onboarding tours, announce new features, or have notification banners in-app.

## Describe the solution you'd like

- Support different modalities (modal, banner, popover)
- Highlight certain areas in a users app, and place the popup next to it
- Page- and rule-based logic, e.g. have a tool tip next to this button, then when they click the button and navigate to another page, highlight another button
- Supports existing survey features, such as user targeting

## Describe alternatives you've considered
- None

## Additional context
- Related ticket, reuqested by user: https://github.com/PostHog/posthog/issues/4879

If you like this idea, please leave a 👍 or ❤️ reaction on this post to vote for it -- your votes and feedback help us prioritize what to work on next!

Please also let us know if there are any important functionalities that are not yet described in the initial list.

**Comment by jovanzulic** [comment]
Hello, this is Joseph and Stephen. After speaking with Scott Sowers, we thought that PostHog was an awesome tool, but we ultimately selected a different analytics vendor since we wanted our all-in-one solution to include a Guides/Walkthroughs feature wherein non-engineers could inject updates into the the front end experience in real time. We have slower release cycles in our B2B scenario, so these real-time injections could help mitigate product inefficiencies much more quickly, as well as get users out of using external training materials.

**Comment by emeril-upsite** [comment]
Whatfix and Shepherd are some alternatives we looked at. Would be great to have this built into PostHog instead

**Comment by samhamad1028** [comment]
This would be incredibly powerful for your B2C customers as most solutions in this space are lacking or far too expensive. One request: minimal native styling with the ability to brand in the client's design system (fonts, colors, etc). 

Command bar's "nudge" tool is a great reference for announcement types

**Comment by jay845** [comment]
So much looking forward to having this. 👍👍

**Comment by ashok-zen** [comment]
Looking forward for this feature. Along with Surveys this will be good for any new product feature launch 👍

**Comment by colttylerwilson** [comment]
Wow... we've been finding a hacky way to use PostHog Survey to do this Product Tour/User Onboarding experience but we've already been looking at Chameleon to get more out of it (still waiting on pushing out Survey's dynamically but we talked to Lucas and it looks like that will soon be in the future).

Anyways, I'm very very very exciting to see this come to light.

**Comment by oos-evercam** [comment]
Very excited for Product Tours in Posthog!

**Comment by juandl** [comment]
Amazing, is thre a timeline for this?

**Comment by nitinmp** [comment]
@jovanzulic can I know which platform you finally go with?

**Comment by gemueselasagne** [comment]
Amazing! We will make use of that for announcing new features to our users. 
Is there any rough timeline for this?

**Comment by IllamosSalaman** [comment]
Would love this feature! I would want an easy way to use my own UI elements to trigger the progression of a product tour. I am using react joyride for my product tour right now, but it really has been a hassle (and also, now broken).

**Comment by mstrausatypical** [comment]
Is this planned in the near future? We would love to have this vs the need to get another product in our stack or build it ourselves.

**Comment by carwei** [comment]
This would be awesome

**Comment by kahunahana** [comment]
Yes, please build product tours. This would be a great addition and reduce the complexity of our stack.

**Comment by Ayusuf95** [comment]
V keen on this feature

**Comment by Armadillidiid** [comment]
It would be great to have this feature as a product. Looking forward to its release!

**Comment by young-snipes** [comment]
It would be a massive add-on to our stake and we would love to be able to do in-app guidance. Is there timeline or roadmap anywhere, so I can view when it's going to happen? 

Cheers.

**Comment by rafleo** [comment]
This would really help migrate from Pendo to Posthog 🤩

**Comment by kcshafer** [comment]
We would definitely use this, currently using posthog but without this we might have to consider other providers.

**Comment by programng** [comment]
Is this currently being worked on, and is there any rough timeline on it? Thank you!

**Comment by phillipmohr** [comment]
Not exactly an all-in-one solution like you’re looking for, but I recently built [https://guidr.us](https://guidr.us) – a no-code product tour widget you can use for free. Might be useful if you’re looking for a standalone way to guide users without writing code. You can fully customize the look and behavior with code.

**Comment by chanmathew** [comment]
Would love to see this added also!

**Comment by xKelvin** [comment]
Would love to see this implemented!

**Comment by darkopia** [comment]
+1 https://posthoghelp.zendesk.com/agent/tickets/41295 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/44169)

**Comment by adboio** [comment]
edit: unfortunately, we have decided not to move forward with product tours as a posthog product.

see more details in [raquel's comment below](https://github.com/PostHog/posthog/issues/25726#issuecomment-4347221420). copy/pasted for LLMs:

> Hey all - we've decided to remove this feature from our roadmap. The reasons are:
> 
> - Tons of other options in the market
> - Low revenue potential
> - Claude / Codex can build bespoke tours very easily, and can hook into our feature flags to help you target them and run experiments on them
> 
> Regarding the latter we're hoping to provide a skill that you can hand off to your agent to get it to make product tours for you.
> 
> We have limited engineering bandwidth as a company, and despite many people being interested in this feature, it didn't feel like the best place to be spending that bandwidth. Thanks all for understanding and I hope you find something great to use for your product tours, if you have recommendations (and may I suggest, skills) to share feel free to do so here

**Comment by younesjallouli** [comment]
Would love to provide feedback, as I'm super interested. There is a deep hole in this market.

All the product tools out there are very, very bad (Pendos and appcues of this world).

First, for startups, they are completely overpriced. Second, even if you're a mid-market/scale-up, their products are very bad. It's mainly because they are made for product managers, and the whole promise is to be able to add announcements and tours with absolutely 0 code. PMs end up inserting CSS selectors, for example, in those tools' dashboards to make things work, which is brittle and just super old school.

The best approach in the market right now is Frigade's (docs.frigade.com). We use it in my company, and it is the best of both worlds:
- Via a dashboard, without technical proficiency:
  - Configure tours content (text, images ...) 
  - Configure user targeting via a dashboard, without technical proficiency
  - See tours/announcements analytics, without technical proficiency
  - Configure cool-offs between tours
- via a Frigade provider in the codebase: All the other stuff (the tour's component, for example, like colors, fonts, button colors ...). It's just a React provider, you don't build the tour by specifying all its subcomponents (think Shadcn, for example), but it's more than enough, we're talking tours and announcements after all. Even the great companies like Cursor and Figma have, look the same at the end of the day.

I like Frigade's product, but unfortunately, it's very buggy. It's not a big team like PostHog. And you feel it in the product experience. Especially when you are developing your tours and need to quickly test things out.

Also, they lack a more modern, and tailwind-native experience. Because this is why their provider configuration ends up looking in my case:

```
    <Frigade.Provider
      apiKey={config.frigadeToken}
      userId={user?.id.toString()}
      userProperties={{
        email: user?.email,
        name: user?.name,
        rollstackCreatedAt: user?.createdAt,
      }}
      theme={{
        colors: {
          primary: {
            border: '#19a0ec',
            surface: '#19a0ec',
            hover: {
              surface: '#0b8ad1',
              border: '#0b8ad1',
            },
          },
          secondary: {
            border: '#d0d5dd',
          },
        },
      }}
      css={{
        '.fr-spotlight': {
          zIndex: 100,
          backgroundColor: 'rgba(0, 0, 0, 0.3)',
        },
        '.fr-announcement': {
          maxWidth: '600px',
          border: '1px solid #d7dce1',
        },
        '.fr-card': {
          padding: '14px',
          border: '1px solid #d7dce1',
          boxShadow: '0px 10px 20px 0px rgb(0 0 0 / 25%)',
        },
        '.fr-button-primary': {
          padding: '5px 5px',
          borderRadius: '6px',
          minWidth: '64px',
          textAlign: 'center',
        },
        '.fr-button-secondary': {
          padding: '5px 5px',
          borderRadius: '6px',
          minWidth: '64px',
          textAlign: 'center',
        },
        '.fr-hint .fr-image, .fr-hint .fr-video': {
          marginLeft: '-14px',
          marginRight: '-14px',
        },
        '.fr-tour .fr-card, .fr-survey .fr-card': {
          backgroundColor: '#0a6fa8',
          border: 'none',
        },
        '.fr-tour .fr-title, .fr-survey .fr-title': {
          color: '#ffffff',
        },
        '.fr-tour .fr-subtitle, .fr-survey .fr-subtitle': {
          color: '#ffffff',
        },
        '.fr-tour .fr-button-primary, .fr-survey .fr-button-primary': {
          backgroundColor: '#ffffff',
          color: '#0a6fa8',
          borderColor: '#ffffff',
        }
      }}
    >
      {children}
    </Frigade.Provider>
```

Happy to provide more feedback.

**Comment by adboio** [comment]
@TravisPageV1 the almighty Hog 🦔 has answered your prayers - shoot me your email or project ID and i'll get you in the early access!

**Comment by raquelmsmith** [comment]
Hey all - we've decided to remove this feature from our roadmap. The reasons are:

- Tons of other options in the market
- Low revenue potential
- Claude / Codex can build bespoke tours very easily, and can hook into our feature flags to help you target them and run experiments on them

Regarding the latter we're hoping to provide a skill that you can hand off to your agent to get it to make product tours for you.

We have limited engineering bandwidth as a company, and despite many people being interested in this feature, it didn't feel like the best place to be spending that bandwidth. Thanks all for understanding and I hope you find something great to use for your product tours, if you have recommendations (and may I suggest, skills) to share feel free to do so here 

<img width="20" height="20" alt="Image" src="https://github.com/user-attachments/assets/b0bc7d57-868d-4510-a973-31f5011304fa" />

**Comment by mareksotak** [comment]
I wanted to share something that may help people here.

At Inline Manual, we’ve created a free Driver.js builder. You can visually build Driver.js tours in our Builder without logging in, then export the configuration and use it directly in your own app.

This fits quite well with the direction PostHog mentioned: use PostHog feature flags, cohorts, experiments, or events to decide when a tour should appear, and use Driver.js as the lightweight product guidance layer, which could be first generated by AI and then maintained/edited by a human.

No account required, free to use, and hopefully useful for teams that want no-code creation while still keeping the tour implementation developer-friendly.

It is a browser extension: https://inlinemanual.com/driverjs/

**Comment by IchordeDionysos** [comment]
@raquelmsmith well the options for remotely configured product tours in Flutter are pretty much non-existent ...

The product tours feature had a announcement banner.

What are the current plana around supporting/expanding in app messages that can be triggered from workflows?

Is this still something PostHog will invest in?
Without those it will be a hard sell to switch from a tool like Braze (which costs us very much) to PostHog.

**Comment by raquelmsmith** [comment]
@marandaneto do we support feature flag payloads in flutter?

If so then @IchordeDionysos I'd recommend creating a banner component in your app that is shown only when a feature flag is enabled for a user, and leverage [feature flag payloads](https://posthog.com/docs/feature-flags/creating-feature-flags#payloads) to control the content of the banner. You can see an example of this where we use payloads to control the content of an upsell component here: 

https://github.com/PostHog/posthog/blob/master/frontend/src/lib/components/NavPanelAdvertisement/NavPanelAdvertisement.tsx#L27-L33 

For triggering form workflows, you can use the `Update person property` feature and then target the flags based on that property.

<img width="426" height="213" alt="Image" src="https://github.com/user-attachments/assets/12b7367d-71b8-4db4-9184-c67c8ee9cb84" />

**Comment by marandaneto** [comment]
yes https://github.com/PostHog/posthog-flutter/blob/969297b5423e7bdafb9a8652966bd5cc9e7085ac/posthog_flutter/lib/src/posthog.dart#L280-L300

**Comment by jchambo** [comment]
> [@TravisPageV1](https://github.com/TravisPageV1) the almighty Hog 🦔 has answered your prayers - shoot me your email or project ID and i'll get you in the early access!

Is this still a thing? Any closer to an official launch?

EDIT: I ended up going with Driver.js for now.
